### PR TITLE
[M] Fix NPE in updateEntitlement (ENT-3887)

### PR DESCRIPTION
**Pull request by JoshAlbrecht23**

### CODE REVIEW

File: src/test/java/org/candlepin/resource/EntitlementResourceTest.java [hunk 1]
```diff
@@ -352,4 +355,103 @@ public void getAllEntitlementsForConsumer() {
...
+        entitlementDTO.setId(entitlementId);
+        entitlementDTO.setQuantity(null);
+
+        doReturn(entitlement).when(entitlementCurator).get(eq(entitlementId));
+
+        entResource.updateEntitlement(entitlementId, entitlementDTO);
+
+        verify(entitler, never()).adjustEntitlementQuantity(any(Consumer.class), any(Entitlement.class), any(Integer.class));
+    }
+    @Test
```
Comment (Ceiu) [inline src/test/java/org/candlepin/resource/EntitlementResourceTest.java:377]: Not sure if checkstyle will catch it, but separate your methods/blocks with whitespace wherever possible.

File: src/main/java/org/candlepin/resource/EntitlementResource.java [hunk 2]
```diff
@@ -186,25 +186,24 @@ public EntitlementDTO getEntitlement(@Verify(Entitlement.class) String entitleme
...
+        }
+
+        Integer updateQuantity = update.getQuantity();
+        if (updateQuantity != null){
+            // Quantity changes must be larger than 0.
+            if (updateQuantity <= 0)
+                throw new BadRequestException(i18n.tr("Quantity value must be greater than 0"));
+            
+            // Ensure a change will occur with the update.
+            if (entitlement.getQuantity() != null && !entitlement.getQuantity().equals(update.getQuantity())) {
```
[the remark this answers is not in the document]
Comment (JoshAlbrecht23) [reply src/main/java/org/candlepin/resource/EntitlementResource.java]: Good call! I updated to use the variable on 195

File: src/test/java/org/candlepin/resource/EntitlementResourceTest.java (hunk 1 above)
Comment (JoshAlbrecht23) [reply src/test/java/org/candlepin/resource/EntitlementResourceTest.java:378]: Added whitespace to separate

File: src/test/java/org/candlepin/resource/EntitlementResourceTest.java [hunk 3]
```diff
@@ -392,12 +392,12 @@ public void testUpdateEntitlementWith0Quantity()
             entResource.updateEntitlement(entitlementId, entitlementDTO)
         );
 
-        verify(entitler, never()).adjustEntitlementQuantity(any(Consumer.class), any(Entitlement.class), any(Integer.class));
+        verify(entitler, never())
+                .adjustEntitlementQuantity(any(Consumer.class), any(Entitlement.class), any(Integer.class));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/resource/EntitlementResourceTest.java:396]: Use 4-space indentation on these continuations

File: src/test/java/org/candlepin/resource/EntitlementResourceTest.java [hunk 4]
```diff
@@ -454,4 +454,18 @@ public void testUpdateEntitlementWithValidQuantityUpdate()
...
     }
+
+    @Test
+    public void testUpdateEntitlementWithInvalidEntitlementId() {
+        String entitlementId = "uniqueId";
+        EntitlementDTO entitlementDTO = new EntitlementDTO();
+        entitlementDTO.setId(entitlementId);
+        entitlementDTO.setQuantity(4);
+
+        doReturn(null).when(entitlementCurator).get(eq(entitlementId));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/resource/EntitlementResourceTest.java:465]: You can omit this entirely for this test, as the default return from a mock is null.

**Review by Ceiu — CHANGES_REQUESTED** [review]
Couple minor changes and a squash and this should be good to go.

File: src/test/java/org/candlepin/resource/EntitlementResourceTest.java (hunk 3 above)
Comment (JoshAlbrecht23) [reply src/test/java/org/candlepin/resource/EntitlementResourceTest.java]: Updated to 4 spaces

File: src/test/java/org/candlepin/resource/EntitlementResourceTest.java (hunk 4 above)
Comment (JoshAlbrecht23) [reply src/test/java/org/candlepin/resource/EntitlementResourceTest.java]: Good call! Updated

**Review by Ceiu — APPROVED** [review]

**Pull request merged**